src/view.rs

```diff
@@ -696,6 +710,30 @@ impl RowsView {
         Ok(())
     }
 
+    pub fn toggle_mark(&mut self, row_index: usize) -> Option<MarkToggleResult> {
+        if let Some(row) = self.rows.get(row_index) {
```

Comment:
Might want to consider using the `?` operator to simplify the code, for example:

```rust
pub fn toggle_mark(&mut self, row_index: usize) -> Option<MarkToggleResult> {
    let record_num = self.rows.get(row_index)?.record_num;

    if self.marked_rows.remove(&record_num) {
        return Some(MarkToggleResult {
            record_num,
            marked: false,
        });
    }

    self.marked_rows.insert(record_num);

    Some(MarkToggleResult {
        record_num,
        marked: true,
    })
}
```

I find the above a bit more readable.